Hit.h

```diff
@@ -174,43 +160,29 @@ class Hit
 
   ~Hit(){}
 
-  const SVector3&     position()   const {return state_.parameters;}
-  const SVector3&     parameters() const {return state_.parameters;}
-  const SMatrixSym33& error()      const {return state_.errors;}
+  const SVector3&  position()  const {return state_.parameters;}
+  const SMatrixSym33& error()  const {return state_.errors;}
+  const SVector3& parameters() const {return state_.parameters;}
+  const float r() const {return std::sqrt(getRad2(state_.parameters.At(0),state_.parameters.At(1)));}
+  const float x() const {return state_.parameters.At(0);}
+  const float y() const {return state_.parameters.At(1);}
+  const float z() const {return state_.parameters.At(2);}
+  const float phi() const {return getPhi(state_.parameters.At(0),state_.parameters.At(1));}
+  const float eta() const {return getEta(r(),z());}
 
   // Non-const versions needed for CopyOut of Matriplex.
   SVector3&     parameters_nc() {return state_.parameters;}
   SMatrixSym33& error_nc()      {return state_.errors;}
 
-  float r() const {
-    return sqrt(state_.parameters.At(0)*state_.parameters.At(0) +
-                state_.parameters.At(1)*state_.parameters.At(1));
-  }
-  float x() const {
-    return state_.parameters.At(0);
-  }
-  float y() const {
-    return state_.parameters.At(1);
-  }
-  float z() const {
-    return state_.parameters.At(2);
-  }
-  float phi() const {
-    return getPhi(state_.parameters.At(0), state_.parameters.At(1));
-  }
-  float eta() const {
-    return getEta(state_.parameters.At(0), state_.parameters.At(1), state_.parameters.At(2));
-  }
-
   const MeasurementState& measurementState() const {
     return state_;
   }
 
   const MCHitInfo& mcHitInfo() const {return mcHitInfo_;}
-  int mcTrackID() const {return mcHitInfo_.mcTrackID_;}
-  int layer() const {return mcHitInfo_.layer_;}
-  int ithLayerHit() const {return mcHitInfo_.ithLayerHit_;}
-  int hitID() const {return mcHitInfo_.mcHitID_;}
+  unsigned int mcTrackID() const {return mcHitInfo_.mcTrackID_;}
+  unsigned int layer() const {return mcHitInfo_.layer_;}
+  unsigned int ithLayerHit() const {return mcHitInfo_.ithLayerHit_;}
+  unsigned int hitID() const {return mcHitInfo_.mcHitID_;}
```

Comment:
I wouldn't go doing a mass conversion since that could introduce bugs (I also generally don't agree with Matevz about unsigned vs. signed (unsigned is simple and forms a group :), but not particularly strongly.

-dan

> On Sep 3, 2015, at 09:58, Kevin McDermott <notifications@github.com> wrote:
> 
> In Hit.h:
> 
> >    const MeasurementState& measurementState() const {
> >      return state_;
> >    }
> >  
> >    const MCHitInfo& mcHitInfo() const {return mcHitInfo_;}
> > -  int mcTrackID() const {return mcHitInfo_.mcTrackID_;}
> > -  int layer() const {return mcHitInfo_.layer_;}
> > -  int ithLayerHit() const {return mcHitInfo_.ithLayerHit_;}
> > -  int hitID() const {return mcHitInfo_.mcHitID_;}
> > +  unsigned int mcTrackID() const {return mcHitInfo_.mcTrackID_;}
> > +  unsigned int layer() const {return mcHitInfo_.layer_;}
> > +  unsigned int ithLayerHit() const {return mcHitInfo_.ithLayerHit_;}
> > +  unsigned int hitID() const {return mcHitInfo_.mcHitID_;}
> 
> @osschar @cerati @dan131riley There are some unsigned ints that are in various places not from my changes.
> 
> Examples: 
> https://github.com/cerati/mictest/blob/devel/ConformalUtils.cc#L21
> https://github.com/cerati/mictest/blob/devel/Propagation.cc#L255-L256
> https://github.com/cerati/mictest/blob/devel/Geometry.h#L20-L22
> Math/ (all over the place)
> 
> Should I change these as well? I feel like ConformalUtils, Propagation, Geometry makes sense, changing Math/ sounds like a bad idea.
> 
> —
> Reply to this email directly or view it on GitHub.
>